app/uk/gov/hmrc/cdsimportsddsfrontend/services/CustomsDeclarationsService.scala

```diff
@@ -40,18 +40,24 @@ class CustomsDeclarationsService @Inject()(appConfig: AppConfig, declarationXml:
   def submit(eori: Eori, declarationViewModel: DeclarationViewModel)(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
     val declaration: Declaration = declarationViewModel.toDeclaration
     val xml = declarationXml.fromImportDeclaration(declaration)
-    submit(eori, xml)
+    val lrn = declarationViewModel.documentationAndReferences.localReferenceNumber
+    submit(eori, xml, lrn)
   }
 
-  def submit(eori: Eori, xml: Elem)(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
+  def submit(eori: Eori, xml: Elem, lrn: Option[String])(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
     httpClient.POSTString[CustomsDeclarationsResponse](
       appConfig.declarationsApi.submitEndpoint, xml.toString(), headers = headers(eori))(responseReader, hc, executionContext)
       .map { customsDeclarationsResponse: CustomsDeclarationsResponse =>
         log.info("Response from Declaration API: " + customsDeclarationsResponse);
+        lrn.map(number => saveDeclaration(eori, number))
```

Comment:
Check with the BAs what they expect to happen if the LRN is missing. Should we really not save the data to the backend? After all an accepted declaration may have be registered with the back end

Reply:
I could but then when we don't have LRN in declaration xml, I could see following exception `[Invalid XML: cvc-pattern-valid: Value '' is not facet-valid with respect to pattern '.*[^\s].*' for type '#AnonType_DeclarationFunctionalReferenceIDType'.`  So, I don't see point of saving the declaration when we have invalid declaration xml.

Reply:
Currently the lrn value is written into the functional reference but won’t the lrn ultimately be written to the TraderAssignedReferenceID field?

Reply:
i think this is just a symptom of our overly-optional form...  LRN isn't an optional field.  Maybe we can just fix this in the form + models as part of this ticket?

Reply:
Maybe good to have another ticket/task to refactor the LRN stuff and anything else not part of posting declaration to backend? Just want to avoid long PR reviews and pushing code in small chunks